app/src/main/kotlin/com/flxrs/dankchat/utils/extensions/ChatItemExtensions.kt

```diff
@@ -35,6 +35,6 @@ fun List<ChatItem>.addAndLimit(
     for (item in collection) {
         if (!checkForDuplications || !this.any { it.message.id == item.message.id })
             add(item)
-        if (size > scrollBackLength) removeAt(0)
+        while (size > scrollBackLength) removeAt(0)
```

Comment:
Don't need this here since we have the `takeLast(scrollBackLength)` at the start